src/app/content/hooks/locationChange/resolveContent.ts

```diff
@@ -67,16 +67,26 @@ const resolveBookReference = async(
   match: Match<typeof content>
 ): Promise<[string, string, string | undefined]> => {
   const state = getState();
-  const bookSlug = match.params.book;
   const currentBook = select.book(state);
 
+  if (!match.params.uuid && !match.params.book) {
+    throw new Error('Nor a book nor a uuid found');
+  }
+
+  const bookSlug = match.params.book
+    ? match.params.book
+    : currentBook && currentBook.id === match.params.uuid
+      ? currentBook.id
+      : match.params.uuid && await osWebLoader.getBookSlugFromId(match.params.uuid);
```

Comment:
pretty sure if you do the union this you won't need `match.params.uuid && ` because ts will know that either params.book or params.uuid will be defined